Code/Framework/AzFramework/AzFramework/Quality/QualityCVarGroup.cpp

```diff
@@ -0,0 +1,244 @@
+        using VisitResponse = AZ::SettingsRegistryInterface::VisitResponse;
+        using Type = AZ::SettingsRegistryInterface::Type;
+
+        if (qualityLevel >= m_numQualityLevels || qualityLevel < 0)
+        {
+            AZ_Warning("QualityCVarGroup",
+                       false,
+                       "Invalid quality level %lld requested for %s which only supports quality levels 0..%lld",
+                       qualityLevel,
+                       m_functor->GetName(),
+                       m_numQualityLevels - 1);
+            return;
+        }
+
+        auto key = FixedValueString::format("%s/Settings", m_path.c_str());
+        auto settingsVisitorCallback = [&](const AZ::SettingsRegistryInterface::VisitArgs& visitArgs)
+        {
+            if (visitArgs.m_type == Type::Object ||
+                visitArgs.m_type == Type::Null ||
+                visitArgs.m_type == Type::NoType)
+            {
+                AZ_Warning("QualityCVarGroup", false, "Invalid setting value type for %.*s, objects and null are not supported, only arrays, bool, string, or numbers.", AZ_STRING_ARG(visitArgs.m_fieldName));
+                return VisitResponse::Skip;
+            }
+
+            AZStd::string_view command = visitArgs.m_fieldName;
+            AZStd::string_view valueKey = visitArgs.m_jsonKeyPath;
+            if (visitArgs.m_type == Type::Array)
+            {
+                valueKey = FixedValueString::format("%.*s/%d", AZ_STRING_ARG(valueKey), qualityLevel);
+                if (visitArgs.m_registry.GetType(valueKey) == Type::NoType)
+                {
+                    // if the array index doesn't exist, use the value for the highest index
+                    ValueType maxQualityLevel = GetHighestQualityForSetting(visitArgs.m_jsonKeyPath);
+                    if (maxQualityLevel < 0)
+                    {
+                        AZ_Warning("QualityCVarGroup", false, "Not applying settings for %.s because the array is empty.", AZ_STRING_ARG(command));
+                        return VisitResponse::Continue;
+                    }
+                    valueKey = FixedValueString::format("%.*s/%d", AZ_STRING_ARG(visitArgs.m_jsonKeyPath), maxQualityLevel);
+                }
+            }
+
+            PerformConsoleCommand(command, valueKey);
+
+            return VisitResponse::Continue;
+        };
+        AZ::SettingsRegistryVisitorUtils::VisitObject(*m_settingsRegistry, settingsVisitorCallback, key);
+    }
+
+    void QualityCVarGroup::PerformConsoleCommand(AZStd::string_view command, AZStd::string_view key)
+    {
+        AZStd::string value;
+        switch (m_settingsRegistry->GetType(key))
+        {
+        case AZ::SettingsRegistryInterface::Type::FloatingPoint:
+            {
+                double floatingPointValue;
+                if (m_settingsRegistry->Get(floatingPointValue, key))
+                {
+                    value = AZStd::to_string(floatingPointValue);

```

Comment:
This can use the [AZ::ConsoleTypeHelpers::ValueToString](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Console/ConsoleTypeHelpers.inl#L23-L35) function overload for each of these cases.

I recommend doing that since it converts bool values to lowercase "true" and "false".

Reply:
I wasn't sure if I wanted to support the object and vector types and just limit this to numbers, bools and strings.  But it's probably going to rare and at least it'll keep our to string logic unified.

Reply:
Well this is more using the same code flow that the AZ::Console uses for converting non-string values for string values to execute in the console.

The code here would still support the restricted types of string, floating point, integer and bool.

Accept it would use the convert to string flow that the Console system is using and has been tested to work with.